### PR TITLE
bpo-33671: Replace platform test for darwin with support.MACOS

### PR DESCRIPTION
**Pull request by 4383**
## Summary
Replace all platform test on darwin system with `support.MACOS` introduced by #7800 

This pull request is focused on `darwin`, `win` will be replaced in another pull request.

https://bugs.python.org/issue33671

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


When your account is ready, please add a comment in this pull request
and a Python core developer will remove the `CLA not signed` label
to make the bot check again.

Thanks again for your contribution, we look forward to reviewing it!

**Comment by 4383** [comment]
Signed CLA :)

File: Lib/test/test_logging.py [hunk 1]
```diff
@@ -547,7 +547,7 @@ def test_name(self):
     def test_builtin_handlers(self):
         # We can't actually *use* too many handlers in the tests,
         # but we can try instantiating them with various options
-        if sys.platform in ('linux', 'darwin'):
+        if MACOS or sys.platform == 'linux':
```
Comment (pablogsal) [inline Lib/test/test_logging.py]: The tests are failing because MACOS is not imported (You need to use `support.MACOS).

**Comment by pablogsal** [comment]
Hi @4383  and thank you for your contribution!

The CI is failing right now with this error:

```exception

ERROR: test_path_objects (test.test_logging.HandlerTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/python/cpython/Lib/test/test_logging.py", line 616, in test_path_objects
    if MACOS or sys.platform == 'linux':
NameError: name 'MACOS' is not defined
```
Notice that to avoid waiting for the CI to fail in the PR you can run your tests locally. For example:

```
./configure --with-pydebug && make && make test
```

**Comment by ned-deily** [comment]
Thanks for the contribution.  But I think this or any similar changes should not be merged until we've further discussed the implications of #7800 itself.

**Comment by 4383** [comment]
Hi @ned-deily, no problems, I can maintain these changes up-to-date during the time you need to measure implications of #7800 :)

Hi @pablogsal, thanks for your help, I pushed too hastily. Errors fixed :)

**Comment by ned-deily** [comment]
It was decided to revert the changes in #7800 so this suggested PR no longer applies.  Thanks again for your contribution, @4383

**Pull request closed without merging**